File: Excel/576 Challenge.py
import pandas as pd

path = "576 Sort only Consonants.xlsx"
input = pd.read_excel(path, usecols="A", nrows=10)
test = pd.read_excel(path, usecols="B", nrows=10)

def process_column(word):
    col = list(word)
    consonant_pos = [i for i, letter in enumerate(col) if letter.lower() in "bcdfghjklmnpqrstvwxyz"]
    sorted_consonants = sorted([col[i] for i in consonant_pos])
    for i, pos in enumerate(consonant_pos):
        col[pos] = sorted_consonants[i]
    return ''.join(col)

input['result'] = input.iloc[:, 0].apply(process_column)

print(input["result"].equals(test["Answer Expected"])) # True

File: Excel/500-599/576/576 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/576 Sort only Consonants.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10")


process_column <- function(word) {
  col <- strsplit(word, "")[[1]]
  consonant_pos <- grep("[b-df-hj-np-tv-z]", col)
  sorted_consonants <- sort(col[consonant_pos])
  col[consonant_pos] <- sorted_consonants
  paste(col, collapse = "") 
}

result = input %>%
  mutate(result = map_chr(Sentences, process_column))
                                                                                      
all.equal(result$result, test$`Answer Expected`, check.attributes = FALSE)
#   [1] TRUE